modules/javafx.web/src/test/java/test/javafx/scene/web/SubresourceIntegrityTest.java

```diff
@@ -90,26 +87,26 @@ public void setup() throws Exception {
                 new File("src/test/resources/test/html/subresource-integrity-test.js").toURI().toASCIIString();
         final String html =
                 String.format("<html>\n" +
-                "<head><script src='%s' integrity='%s' crossorigin='anonymous'></script></head>\n" +
-                "<body>%s</body>\n" +
-                "</html>", scriptUrl, hashValue, NOT_LOADED);
+                        "<head><script src='%s' integrity='%s' crossorigin='anonymous'></script></head>\n" +
+                        "<body>%s</body>\n" +
+                        "</html>", scriptUrl, hashValue, NOT_LOADED);
         out.write(html.getBytes());
         out.close();
     }
 
-    @Test
+    @ParameterizedTest
+    @MethodSource("dataProvider")
     public void testScriptTagWithCorrectHashValue() {
```

Comment:
Major: This methods need the parameters from the `MethodSource` AFAIK, that is:
`testScriptTagWithCorrectHashValue(String hashValue, String expected) { .. }`

The member variables should be removed then:
``` 
private final String hashValue = "";
private final String expected = "";
```

Reply:
In this case, keeping the values as instance variables is more reliable. Since all the tests pass when hashValue and expected are instance variables and  some of them fail when I  adding parameter to testScriptTagWithCorrectHashValue , 

it's likely that their assignment happens at the correct time relative to the @BeforeEach setup. Using parameters may interfere with the test lifecycle, since hasValue is being used in 
    @BeforeEach
    public void setup() ....{}

 it's safer and more straightforward to use instance variables.This would also make it easier to manage the state across @BeforeEach

Reply:
I think you need to do the same as done by @andy-goryachev-oracle in his JUnit5 replacement PR:
```
    public void setup(String hashValue, String expected) throws Exception {
        htmlFile = new File("subresource-integrity-test.html");
        ...
    }

    @ParameterizedTest
    @MethodSource("dataProvider")
    public void testScriptTagWithCorrectHashValue(String hashValue, String expected) {
        setup(hashValue, expected);
        ...
    }
```

Reply:
@Maran23 is right.  In the absence of support for parameterized class-level tests the easiest solution (I think) is to parameterize every `@Test`, which means:
- add parameters to each method, replace `@Test` with `@ParameterizedTest` and `@MethodSource`
- add parameters to setup method (`@Before`, remove the annotation)
- call the setup method from each test method
- remove the fields that were set in the constructor and the constructor itself

Things might be a bit more complicated in the case of test class hierarchies, so additional work and checks will be needed.

Reply:
When hashValue and expected are instance variables, they are initialized once per test case due to how JUnit creates a new instance of the test class for each @ParameterizedTest. This ensures that the state is clean between tests and no variable is shared between different test cases. So I think it is ok.

Reply:
I also think the test needs to be modified as suggested by @Maran23 and @andy-goryachev-oracle .
With the changes in this PR the member variables are never assigned with value.
I verified the values of the two member variables by printing them before line 102, They are always empty.
Apply below change, after applying this PR changes.
```
diff --git a/modules/javafx.web/src/test/java/test/javafx/scene/web/SubresourceIntegrityTest.java b/modules/javafx.web/src/test/java/test/javafx/scene/web/SubresourceIntegrityTest.java
index e65ff7d32f..0d2d670777 100644
--- a/modules/javafx.web/src/test/java/test/javafx/scene/web/SubresourceIntegrityTest.java
+++ b/modules/javafx.web/src/test/java/test/javafx/scene/web/SubresourceIntegrityTest.java
@@ -99,6 +99,8 @@ public final class SubresourceIntegrityTest extends TestBase {
         load(htmlFile);
         final String bodyText = (String) executeScript("document.body.innerText");
         assertNotNull("document.body.innerText must be non null for " + hashValue, bodyText);
+        System.err.println("hashValue = " + hashValue);
+        System.err.println("expected = " + expected);
         assertEquals(hashValue, expected, bodyText);
     }
```

<img width="638" alt="Screenshot 2024-09-17 at 4 51 46 PM" src="https://github.com/user-attachments/assets/5dd4360f-a82d-46b6-8828-3406d63a214f">

Reply:
@jaybhaskar Yes, there is an issue as the above comment from @arapte shows. Please don't mark this as "resolved" until it is, and @arapte is satisfied.

Reply:
Thanks , I have modified and push according

Reply:
I found issue , and done changes according to review, Thanks all